tests/subsys/bluetooth/mesh/scheduler_model/action_planning/src/main.c

```diff
@@ -0,0 +1,603 @@
+		.year = 10,
+		.month = ANY_MONTH,
+		.day = 1,
+		.hour = 0,
+		.minute = 0,
+		.second = 30,
+		.day_of_week = ANY_DAY_OF_WEEK,
+		.action = BT_MESH_SCHEDULER_TURN_ON,
+		.transition_time = model_transition_encode(2500),
+		.scene_number = 1,
+	};
+
+	struct sched_evt evt_list[] = {
+		{ .elem_idx = 0, .evt = BT_MESH_SCHEDULER_TURN_ON, .transition_time = 2500 },
+		{ .elem_idx = 1, .evt = BT_MESH_SCHEDULER_TURN_ON, .transition_time = 2500 },
+		{ .elem_idx = 2, .evt = BT_MESH_SCHEDULER_TURN_ON, .transition_time = 2500 },
+	};
+
+	sched_evt_ctx_init(evt_list, ARRAY_SIZE(evt_list));
+
+	action_put(&test_action, sched_mod_elem1);
+	measurement_start(60, ARRAY_SIZE(evt_list));
+
+	struct tm expected[3] = {
+		{ ISTM(110, 0, 1, 0, 0, 30) },
+		{ ISTM(110, 0, 1, 0, 0, 30) },
+		{ ISTM(110, 0, 1, 0, 0, 30) },
+	};
+
+	expected_tm_check(fired_tm, expected, 1);
+}
+
+static void test_second_sched_turn_off(void)
+{
+	/** Send a TURN_OFF action to the sched srv in
+	 * element 4 of the mocked composition data.
+	 *
+	 * Expect the onoff srv in elem 4 and 5 to trigger
+	 * in sequence. Also verifies the correctness
+	 * of the injected parameters and the number and
+	 * timing of events.
+	 */
+
+	const struct bt_mesh_schedule_entry test_action = {
+		.year = 10,
+		.month = ANY_MONTH,
+		.day = 1,
+		.hour = 0,
+		.minute = 0,
+		.second = 30,
+		.day_of_week = ANY_DAY_OF_WEEK,
+		.action = BT_MESH_SCHEDULER_TURN_OFF,
+		.transition_time = model_transition_encode(100),
+		.scene_number = 1,
+	};
+
+	struct sched_evt evt_list[] = {
+		{ .elem_idx = 3, .evt = BT_MESH_SCHEDULER_TURN_OFF, .transition_time = 100 },
+		{ .elem_idx = 4, .evt = BT_MESH_SCHEDULER_TURN_OFF, .transition_time = 100 },
+	};
+
+	sched_evt_ctx_init(evt_list, ARRAY_SIZE(evt_list));
+
+	action_put(&test_action, sched_mod_elem4);
+	measurement_start(60, ARRAY_SIZE(evt_list));
+
+	struct tm expected[2] = {
+		{ ISTM(110, 0, 1, 0, 0, 30) },
+		{ ISTM(110, 0, 1, 0, 0, 30) },
+	};
+
+	expected_tm_check(fired_tm, expected, 1);
+}
+
+static void test_second_sched_scene_recall(void)
+{
+	/** Send a SCENE_RECALL action to the sched srv in
+	 * element 4 of the mocked composition data.
+	 *
+	 * Expect the scene srv in elem 4 and 5 to trigger
+	 * in sequence. Also verifies the correctness
+	 * of the injected parameters and the number and
+	 * timing of events.
+	 */
+
+	const struct bt_mesh_schedule_entry test_action = {
+		.year = 10,
+		.month = ANY_MONTH,
+		.day = 1,
+		.hour = 0,
+		.minute = 0,
+		.second = 30,
+		.day_of_week = ANY_DAY_OF_WEEK,
+		.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+		.transition_time = model_transition_encode(500),
+		.scene_number = 4,
+	};
+
+	struct sched_evt evt_list[] = {
+		{
+			.elem_idx = 3,
+			.evt = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 500,
+			.scene_number = 4,
+		},
+		{
+			.elem_idx = 4,
+			.evt = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 500,
+			.scene_number = 4,
```

Comment:
Could you use the different scene number?

Reply:
You mean other than 4?

Reply:
yep, to be sure it isn't leftover the previous event.

Reply:
As far as I can see this is the only test where scene number 4 is used. Do you mean for the different events in this particular test? In that case I don't think that is possible? Since the test only has one scheduler event happening (with scene number = 4) that will be received once on each of the scene servers on element 4 and 5.

Reply:
I meant that it should check that the scheduler model provide the new scene number for the new scene. Using different digits gives assurance that sequentially is happens correctly. There is no leftover of the previous scene handler.

Reply:
Discussed IRL. This is Ok.